src/librustc_mir/hair/cx/expr.rs

```diff
@@ -75,19 +75,19 @@ fn apply_adjustment<'a, 'gcx, 'tcx>(cx: &mut Cx<'a, 'gcx, 'tcx>,
                                     -> Expr<'tcx> {
     let Expr { temp_lifetime, mut span, .. } = expr;
     let kind = match adjustment.kind {
-        Adjust::ReifyFnPointer => {
+        Adjust::Pointer(PointerCast::ReifyFnPointer) => {
             ExprKind::ReifyFnPointer { source: expr.to_ref() }
```

Comment:
All of these `ExprKind`s created from `Adjust::Pointer` could be unified into one variant that has a `PointerCast` field and a `source` field.

Reply:
Yeah I will refactor `ExprKind`.